Makefile

```
@@ -210,8 +210,8 @@ SEMCONVPKG ?= "semconv/"
 semconv-generate: | $(SEMCONVGEN) $(SEMCONVKIT)
 	@[ "$(TAG)" ] || ( echo "TAG unset: missing opentelemetry specification tag"; exit 1 )
 	@[ "$(OTEL_SPEC_REPO)" ] || ( echo "OTEL_SPEC_REPO unset: missing path to opentelemetry specification repo"; exit 1 )
-	@$(SEMCONVGEN) -i "$(OTEL_SPEC_REPO)/semantic_conventions/trace" -t "$(SEMCONVPKG)/template.j2" -s "$(TAG)"
-	@$(SEMCONVGEN) -i "$(OTEL_SPEC_REPO)/semantic_conventions/resource" -t "$(SEMCONVPKG)/template.j2" -s "$(TAG)"
+	@$(SEMCONVGEN) -i "$(OTEL_SPEC_REPO)/semantic_conventions/." --only=span -p conventionType=trace -f trace.go -t "$(SEMCONVPKG)/template.j2" -s "$(TAG)"
+	@$(SEMCONVGEN) -i "$(OTEL_SPEC_REPO)/semantic_conventions/." --only=resource -p conventionType=resource -f resource.go -t "$(SEMCONVPKG)/template.j2" -s "$(TAG)"
```

```suggestion
	@$(SEMCONVGEN) -i "$(OTEL_SPEC_REPO)/semantic_conventions" --only=span -f trace.go -t "$(SEMCONVPKG)/template.j2" -s "$(TAG)"
	@$(SEMCONVGEN) -i "$(OTEL_SPEC_REPO)/semantic_conventions" --only=resource -f resource.go -t "$(SEMCONVPKG)/template.j2" -s "$(TAG)"
```

I don't believe we use `conventionType` in our template, so no need to include that in the invocation.

---

Without the `-p`:

```terminal
panic: unable to render template: exit status 2
usage: gen-semconv [-h] [--debug]
                   [--only {span,resource,event,metric,units,scope,attribute_group}]
                   [--yaml-root folder] [--exclude EXCLUDE]
                   [YAML_FILE [YAML_FILE ...]] {code,markdown} ...
gen-semconv: error: No semantic convention model found!


goroutine 1 [running]:
main.main()
	/home/tyler/go/src/go.opentelemetry.io/build-tools/semconvgen/generator.go:62 +0x48e
make: *** [Makefile:213: semconv-generate] Error 2
```

---

interesting.  v0v

---

I'll look further into this tomorrow.

---

Yes that is caused by this join: https://github.com/open-telemetry/opentelemetry-go-build-tools/blob/v0.3.0/semconvgen/generator.go#L167.
It will take the last part of the provided path and add it to a pre defined one. Adding a `.` at  the end of the path allow us to avoid a breaking change.

---

@frzifus where did you pull the `-p conventionType=*` lines from? I can't seem to find reference to this in the `build-tools`, the semantic convention generation python scripts, the yaml parse, or the semantic conventions themselves.